File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # create an empty hash table
        ht = {}
        
        # loop through each element in the array
        for i in range(len(nums)):
            # calculate the complement of the current element
            complement = target - nums[i]
            
            # check if the complement exists in the hash table
            if complement in ht:
                # return the indices of the current element and its complement
                return [ht[complement], i]
            
            # otherwise, add the current element and its index to the hash table
            ht[nums[i]] = i
        
        # if no solution is found, return an empty array
        return []

File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {

       Dictionary<int, int> dict = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++)
        {
            if (dict.ContainsKey(target - nums[i]))
            {
                return new int[] { dict[target - nums[i]], i };
            }
            else
            {
                dict[nums[i]] = i;
            }
        }
        return new int[0];
    }
}